feature-libs/product/configuration/src/textfield-configurator/state/effects/configurator-textfield.effect.ts

```diff
@@ -0,0 +1,143 @@
+import { Injectable } from '@angular/core';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import { CartActions, GenericConfigurator } from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { catchError, map, mergeMap, switchMap } from 'rxjs/operators';
+import { makeErrorSerializable } from '../../../utils/serialization-utils';
+import { ConfiguratorTextfieldConnector } from '../../connectors/configurator-textfield.connector';
+import { ConfiguratorTextfield } from '../../model/configurator-textfield.model';
+import { ConfiguratorTextfieldActions } from '../actions/index';
+@Injectable()
+export class ConfiguratorTextfieldEffects {
+  @Effect()
+  createConfiguration$: Observable<
+    | ConfiguratorTextfieldActions.CreateConfiguration
+    | ConfiguratorTextfieldActions.CreateConfigurationSuccess
+    | ConfiguratorTextfieldActions.CreateConfigurationFail
+  > = this.actions$.pipe(
+    ofType(ConfiguratorTextfieldActions.CREATE_CONFIGURATION),
+    map(
+      (action: ConfiguratorTextfieldActions.CreateConfiguration) =>
+        action.payload
+    ),
+    mergeMap((payload) => {
```

Comment:
If multiple `CreateConfiguration` actions are dispatched, using `mergeMap` here will process all of them in parallel, which is not necessarily a bad thing, just bringing it up.
If a new CreateConfiguration action comes down the pipe, you might want to:
- cancel the currently processing action? use `switchMap`
- queue the new action after the current one? use `concatMap`
- ignore the new action, while the current one is still being processed? use `exchaustMap`

Reply:
The same applies to other effects.

Reply:
Thanks, correct, we missed this. Should be mostly switchMap